Shokofin/Providers/Helper.cs

```diff
@@ -15,100 +18,180 @@ public static string GetImageUrl(Image image)
             return image != null ? $"http://{Plugin.Instance.Configuration.Host}:{Plugin.Instance.Configuration.Port}/api/v3/Image/{image.Source}/{image.Type}/{image.ID}" : null;
         }
 
+        public static (int, int) GetNumbers(Models.Series series, Models.Episode.AniDB episode)
+        {
+            return (GetIndexNumber(series, episode), GetMaxNumber(series));
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="series"></param>
+        /// <param name="episode"></param>
+        /// <returns></returns>
+        public static int GetIndexNumber(Models.Series series, Models.Episode.AniDB episode)
+        {
+            int offset = 0;
+            switch (episode.Type)
+            {
+                case EpisodeType.Episode:
+                    break;
+                case EpisodeType.Special:
+                    offset += series.Sizes.Total.Episodes;
+                    break; // goto case EpisodeType.Episode;
+                case EpisodeType.Credits:
+                    offset += series.Sizes.Total?.Specials ?? 0;
+                    goto case EpisodeType.Special;
+                case EpisodeType.Other:
+                    offset += series.Sizes.Total?.Credits ?? 0;
+                    goto case EpisodeType.Credits;
+                case EpisodeType.Parody:
+                    offset += series.Sizes.Total?.Others ?? 0;
+                    goto case EpisodeType.Other;
+                case EpisodeType.Trailer:
+                    offset += series.Sizes.Total?.Parodies ?? 0;
+                    goto case EpisodeType.Parody;
+            }
+            return offset + episode.EpisodeNumber;
+        }
+
+        public static int GetMaxNumber(Models.Series series)
+        {
+            var dict = series.Sizes.Total;
+            return dict.Episodes + dict?.Specials ?? 0 + dict?.Credits ?? 0 + dict?.Others ?? 0 + dict?.Parodies ?? 0 + dict?.Trailers ?? 0;
+        }
+
+        public static ExtraType? GetExtraType(Models.Episode.AniDB episode)
+        {
+            switch (episode.Type)
+            {
+                case EpisodeType.Episode:
+                    return null;
+                case EpisodeType.Trailer:
+                    return ExtraType.Trailer;
+                case EpisodeType.Special: {
+                    var enTitle = Helper.GetTitleByLanguages(episode.Titles, "en");
+                    if (enTitle != null && (enTitle.Contains("intro") || enTitle.Contains("outro"))) {
+                        return ExtraType.DeletedScene;
+                    }
+                    return ExtraType.Scene;
+                }
+                default:
+                    return null;
+            }
+        }
+
+        public static int GetFlagFilter()
```

Comment:
Can you rename that to `GetTagFilter`? I named it wrong and it not very clear what it is.

Reply:
done.